libs/framework/tst/bundle_context_services_test.cpp

```diff
@@ -87,7 +87,7 @@ TEST(CelixBundleContextServicesTests, registerMultipleAndUseServices) {
     };
 
     const char *calcName = "calc";
-    struct calc svc;
+    struct calc svc{};
```

Comment:
This generates an error. Add {nullptr} initialisation to (function pointer) field to prevent this.

Reply:
Do you mean a compiler error? This online example shows that it properly initializes the struct with 0: http://cpp.sh/8xalo

Reply:
So the error is actually an error of the g++ 4.x and 5.x series. The warning is used to indicate class types that are improperly initialized. Since c++11, empty initialization means [zero initialization for non-union class types](https://en.cppreference.com/w/cpp/language/zero_initialization) (as is the case here). As this does not lead to half-initialized  Therefore, since version 6.1 ([godbolt](https://godbolt.org/z/T6yuJD)), g++ includes the following exception for this warning:

`in C++ this option does not warn about the empty { } initializer` [[docs]](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)

As it is a hassle to not apply this rule for these few instances, because it is test code and every time the calc field is properly filled, I will simply revert the changes here.